lib/matplotlib/axis.py

```diff
@@ -1643,6 +1643,19 @@ def set_ticklabels(self, ticklabels, *args, minor=False, **kwargs):
         # replace the ticklabels list with the processed one
         ticklabels = get_labels
 
+        get_labels = []
+        if(rcParams['text.usetex'] and
```

Comment:
Remember that text objects can each have their individual usetex setting, the global setting is just a default.  So it's the text's usetex value that needs to be checked.